src/epoch_credits/mod.rs

```diff
@@ -1 +1,35 @@
+use anyhow::anyhow;
+use solana_client::rpc_client::RpcClient;
+use solana_sdk::clock::Epoch;
+use solana_transaction_status::Rewards;
+
 pub mod caching;
+pub mod caching_metadata;
+
+const SLOT_OFFSET: u64 = 20;
```

Comment:
Not entirely convinced there couldn't be 20 or more skipped slots in a row at the start of an epoch. I think you could try more blocks, possibly in several iterations.

Can you try `SLOT_OFFSET = 1000` or something like that? The RPC call and result wouldn't be too big.

Reply:
I tried `SLOT_OFFSET = 1000` but with `curl` the request took nearly 4 seconds to fulfil using https://api.mainnet-beta.solana.com. Would something like 100 be more appropriate?

Reply:
In that case, sure, 100 or maybe a few requests with smaller intervals would be better.

Reply:
Following up on this - `SLOT_OFFSET` is still 20. Does it need to be changed?

Reply:
Even just doubling it up to 40 would give us plenty of margin. I think 40 would be good.

Reply:
Fixed in 53207a5.